sonar-scanner-engine/src/main/java/org/sonar/batch/report/CoveragePublisher.java

```diff
@@ -62,25 +62,35 @@ public void apply(String value, LineCoverage.Builder builder) {
             builder.setUtHits(Integer.parseInt(value) > 0);
           }
         });
-      applyLineMeasure(resource.key(), lineCount, CoreMetrics.CONDITIONS_BY_LINE_KEY, coveragePerLine,
+      applyLineMeasure(resource.key(), lineCount, CoreMetrics.IT_COVERAGE_LINE_HITS_DATA_KEY, coveragePerLine,
+        new MeasureOperation() {
+          @Override
+          public void apply(String value, LineCoverage.Builder builder) {
+            builder.setItHits(Integer.parseInt(value) > 0);
+          }
+        });
+      // Just in case we have only IT and no UT, we first take conditions from ITs, then from UTs (UTs will override ITs).
+      // Note that it would be very strange (and unsupported in scanner report) to have ITs and UTs don't report the same number of
+      // conditions. Not even talking about overall conditions...
+      applyLineMeasure(resource.key(), lineCount, CoreMetrics.IT_CONDITIONS_BY_LINE_KEY, coveragePerLine,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "IT_CONDITIONS_BY_LINE_KEY"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)